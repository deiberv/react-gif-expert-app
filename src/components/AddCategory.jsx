import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('');
  
  const onInputChange = ({target}) => {
    setInputValue(target.value);
  }

  const onSubmit = (event) => {
    event.preventDefault();
    const newCategory = inputValue.trim();
    if ( newCategory.length <=1 ) return;

    onNewCategory( newCategory );
    setInputValue('');

  }

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input 
        type="text"
        className="form-control my-3"
        value={inputValue}
        onChange={onInputChange}
      />    
    </form>
    
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}