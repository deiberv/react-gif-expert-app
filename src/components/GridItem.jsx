import PropTypes from 'prop-types'

export const GridItem = ({title, url, id}) => {
  return (
    <div className="col-3 col-md-4 col-sm-6 flex-column justify-content-center">
      <img src={ url } alt={ title } className="img-fluid"/>
      <p> { title} </p>
    </div>
  )
}

GridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url  : PropTypes.string.isRequired
}