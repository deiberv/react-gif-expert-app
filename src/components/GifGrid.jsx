

import { GridItem } from "./GridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { Loading } from "../helpers/loading";

export const GifGrid = ({ category }) => {
  
  const { images, isLoading } = useFetchGifs(category);
  
  return (
    <div className="card mb-3">
      <div className="card-header"> 
        <h3> { category }</h3>
      </div>
      <div className="card-body">
        <Loading isLoading={isLoading} />
        <div className="row flex-row justify-content-center">
          { images.map( (image) => (
            <GridItem key={image.id} 
              { ...image} 
            />
          ))}
        </div>
      </div>
    </div>
  )
}
