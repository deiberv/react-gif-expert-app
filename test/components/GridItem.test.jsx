import { render, screen } from "@testing-library/react"
import { GridItem } from "../../src/components/GridItem"

describe('Pruebas en GridItem component', () => {
  const title = 'Saitama';
  const url   = 'https://one-ponch.com/saitama.jpg';

  test('debe hacer match con snapshop', () => { 
    const {container} = render(<GridItem title={ title } url={ url } />);
    expect(container).toMatchSnapshot();
  })

  test('Debe contener url y alt', () => { 
    render(<GridItem title={ title } url={ url } />);
    const { src, alt } =  screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
   })

   test('Debe contener titulo', () => { 
    render(<GridItem title={ title } url={ url } />);
    const { src, alt } =  screen.getByRole('img');
    expect(screen.getByText(title)).toBeTruthy();
   })
})