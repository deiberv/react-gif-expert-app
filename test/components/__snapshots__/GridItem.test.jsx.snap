// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GridItem component debe hacer match con snapshop 1`] = `
<div>
  <div
    class="col-3 col-md-4 col-sm-6 flex-column justify-content-center"
  >
    <img
      alt="Saitama"
      class="img-fluid"
      src="https://one-ponch.com/saitama.jpg"
    />
    <p>
       
      Saitama
       
    </p>
  </div>
</div>
`;
